src/plugins/condition/components/Criterion.vue

```diff
@@ -0,0 +1,145 @@
+                    @change="updateOperationInput"
+            >
+                <option value="">- Select Comparison -</option>
+                <option v-for="option in operations"
+                        :key="option.name"
+                        :value="option.name"
+                >
+                    {{ option.text }}
+                </option>
+            </select>
+            <input v-if="isInputOperation"
+                   v-model="comparisonInputValue"
+                   class="t-condition-name-input"
+                   type="text"
+                   @change="persist"
+            >
+        </span>
+    </span>
+</li>
+
+</template>
+
+<script>
+import { OPERATIONS } from '../utils/operations';
+
+export default {
+    inject: ['openmct'],
+    props: {
+        criterion: {
+            type: Object,
+            required: true
+        },
+        telemetry: {
+            type: Array,
+            required: true,
+            default: () => []
+        },
+        index: {
+            type: Number,
+            required: true
+        },
+        condition: {
+            type: Object,
+            required: true
+        },
+        trigger: {
+            type: String,
+            required: true
+        }
+    },
+    data() {
+        return {
+            telemetryMetadata: {},
+            operations: OPERATIONS,
+            selectedTelemetryObject: '',
+            selectedFieldOption: {},
+            selectedOperationOption: '',
+            operationValue: '',
+            comparisonInputValue: '',
+            isInputOperation: false,
+            rowLabel: ''
+        }
+    },
+    computed: {
+        setRowLabel: function () {
+            let operator = this.trigger === 'all' ? 'and ': 'or ';
+
+            return (this.index !== 0 ? operator : '') + 'when';
+        }
+    },
+    mounted() {
+        this.initialize();
+    },
+    updated() {
+        this.persist();
+    },
+    methods: {
+        initialize() {
+            this.selectedTelemetryObject = this.criterion.telemetry;
+            this.selectedFieldOption = this.criterion.metadata;
+            this.selectedOperationOption = this.criterion.operation;
+            this.comparisonInputValue = this.criterion.input;
+            this.updateOperationInput();
+            this.updateFieldOptions();
+
+        },
+        updateFieldOptions() {
+            if (this.selectedTelemetryObject) {
+                this.openmct.objects.get(this.selectedTelemetryObject).then((telemetryObject) => {
+                    this.telemetryMetadata = this.openmct.telemetry.getMetadata(telemetryObject).values();
+                });
+            }
+        },
+        updateOperationInput() {
+            if (this.selectedOperationOption &&
+                (this.selectedOperationOption !== 'isUndefined' &&
+                 this.selectedOperationOption !== 'isDefined')) {
+                this.isInputOperation = true;
```

Comment:
Instead of using specific operation names here like 'isUndefined' and 'isDefined', you should use the operation's `inputCount` to show or hide the input field. That way if more operations are added in the future which don't need an input you won't have to manually add the name of that operation here.
See here: https://github.com/nasa/openmct/blob/24bbcb466ff6839ea7b31d0278daacb95d1b3af5/src/plugins/condition/utils/operations.js#L165

Reply:
done